frontend/lib/src/components/widgets/DataFrame/ColumnMenu.tsx

```diff
@@ -0,0 +1,194 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2025)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { memo, ReactElement, useEffect } from "react"
+
+import { useTheme } from "@emotion/react"
+import { ACCESSIBILITY_TYPE, PLACEMENT, Popover } from "baseui/popover"
+
+import {
+  EmotionTheme,
+  hasLightBackgroundColor,
+} from "@streamlit/lib/src/theme"
+import { DynamicIcon } from "@streamlit/lib/src/components/shared/Icon"
+
+import { StyledMenuList, StyledMenuListItem } from "./styled-components"
+
+export interface ColumnMenuProps {
+  // The top position of the menu
+  top: number
+  // The left position of the menu
+  left: number
+  // Callback to close the menu
+  menuClosed: () => void
```

Comment:
nit: I would name this `onMenuClosed` which sounds a little bit more common

Reply:
Renamed 👍  And also renamed the second callback, see the comment below.